### PR TITLE
Introduce combined elementwise ops

### PR DESCRIPTION
**Pull request by bartekxk**
- Introduce combined elementwise ops
- Extend elementwise ops

### CODE REVIEW

File: example/44_elementwise_permute/elementwise_binary_4D_fp16.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    128,                             // M0PerBlock
+    128,                             // M1PerBlock
+    8,                               // M0PerThread
+    8,                               // M1PerThread
+    ck::Sequence<1, 0>,              // ThreadClusterArrangeOrder
+    ck::Sequence<8, 8>,              // InScalarPerVectorSeq
+    ck::Sequence<8>>;                // OutScalarPerVectorSeq
+
+template <typename HostTensorA0, typename HostTensorA1, typename HostTensorB>
+void host_elementwise4D(
```
Comment (geyyer) [inline example/44_elementwise_permute/elementwise_binary_4D_fp16.cpp]: Would it make sense to separate host calculation as we do here https://github.com/ROCm/composable_kernel/blob/develop/library/include/ck/library/reference_tensor_operation/cpu/reference_gemm.hpp

File: example/44_elementwise_permute/elementwise_permute_4D_fp16_col.cpp [hunk 2]
```diff
@@ -36,9 +39,13 @@ using DeviceElementwisePermuteInstance = ck::tensor_operation::device::DeviceEle
     ck::Sequence<8>,      // InScalarPerVectorSeq
     ck::Sequence<8>>;     // OutScalarPerVectorSeq
 
-template <typename HostTensorA, typename HostTensorB, typename Functor>
-void host_elementwise4D(HostTensorB& B_nhwc, const HostTensorA& A_nchw, Functor functor)
+template <typename HostTensorA, typename HostTensorB>
+void host_elementwise4D(HostTensorB& B_nhwc, const HostTensorA& A_nchw, float scale)
```
Comment (geyyer) [inline example/44_elementwise_permute/elementwise_permute_4D_fp16_col.cpp]: Same here and below

File: include/ck/tensor_operation/gpu/element/binary_element_wise_operation.hpp [hunk 3]
```diff
@@ -92,6 +92,110 @@ struct Add
...
+    operator()<half_t>(half_t& y, const half_t& x0, const half_t& x1) const
+    {
+        y = x0 * x1;
+    };
+
+    template <>
+    __host__ __device__ constexpr void
+    operator()<float>(float& y, const float& x0, const bhalf_t& x1) const
+    {
+        const float x1_tmp = ck::type_convert<float>(x1);
```
Comment (geyyer) [inline include/ck/tensor_operation/gpu/element/binary_element_wise_operation.hpp:168]: Do we need a separate const variable here and below?

File: include/ck/utility/math_v2.hpp [hunk 4]
```diff
@@ -111,6 +115,276 @@ inline __host__ double tanh<double>(double x)
...
+template <>
+inline __host__ double floor<double>(double x)
+{
+    return std::floor(x);
+};
+
+template <typename T>
+inline __host__ T rcp(T x)
+{
+    return ck::type_convert<T>(1.f / ck::type_convert<float>(x));
```
Comment (geyyer) [inline include/ck/utility/math_v2.hpp:385]: Should we take care of division by zero here?

**Review by geyyer — COMMENTED** [review]
@bartekxk, the PR looks great, have a couple of comments below.

File: include/ck/utility/math_v2.hpp (hunk 4 above)
Comment (bartekxk) [reply include/ck/utility/math_v2.hpp:385]: I'm not sure how to handle it. Reciprocal of 0 is not defined. So user probably should be aware of usage rcp.

File: include/ck/tensor_operation/gpu/element/binary_element_wise_operation.hpp (hunk 3 above)
Comment (bartekxk) [reply include/ck/tensor_operation/gpu/element/binary_element_wise_operation.hpp:168]: Just to increase code clarity.

File: example/44_elementwise_permute/elementwise_permute_4D_fp16_col.cpp (hunk 2 above)
Comment (bartekxk) [reply example/44_elementwise_permute/elementwise_permute_4D_fp16_col.cpp]: Done.

File: example/44_elementwise_permute/elementwise_binary_4D_fp16.cpp (hunk 1 above)
Comment (bartekxk) [reply example/44_elementwise_permute/elementwise_binary_4D_fp16.cpp]: Done.

**Review by geyyer — APPROVED** [review]
LGTM!

**Review by zjing14 — APPROVED** [review]

**Pull request merged**